app/controllers/ReportingController.scala

```diff
@@ -120,6 +130,13 @@ trait ReportingController extends BaseController {
     }
   }
 
+  private def enrichReport(candidates: List[CandidateProgressReportItem], postcodes: Map[String, String]):
+  Future[List[CandidateProgressReportItem]] = {
+
+    Future(candidates.map(candidate => candidate.copy(
```

Comment:
The thread is already a pool thread, and it is not done in parallel either, so it doesn't seem to be provide any benefit.  Have you benchmarked this for performance? If there is a performance issue, it may be better to pass the location lookup map into `reportRepository.candidateProgressReport` so that no copying is required